test/sequential/feeLock.API.feePayment.test.ts

```diff
@@ -0,0 +1,399 @@
+  await addMgaToWhitelisted();
+
+  await testUser1.addMGATokens(sudo);
+  testUser1.addAsset(MGA_ASSET_ID);
+
+  await checkErrorSellAsset(
+    "1010: Invalid Transaction: The swap prevalidation has failed",
+    toBN("1", 26),
+    false
+  );
+});
+
+test("gassless- For low-value swaps, token reservation status and pallet storage are altered in accordance with the timeout mechanism", async () => {
+  let accountFeeLockAmount: any;
+  const api = getApi();
+
+  async function checkAccountFeeLockAmount(value: any) {
+    accountFeeLockAmount = JSON.parse(
+      JSON.stringify(
+        await api.query.feeLock.accountFeeLockData(
+          testUser1.keyRingPair.address
+        )
+      )
+    ).totalFeeLockAmount;
+    expect(new BN(accountFeeLockAmount)).bnEqual(new BN(value));
+  }
+
+  await addMgaToWhitelisted();
+
+  await testUser1.addMGATokens(sudo);
+  testUser1.addAsset(MGA_ASSET_ID);
+  await checkAccountFeeLockAmount(0);
+
+  const sellAssetsValue = thresholdValue.sub(new BN(5));
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+  await testUser1.sellAssets(MGA_ASSET_ID, firstCurrency, sellAssetsValue);
+  await waitForNBlocks(periodLength);
+  await checkAccountFeeLockAmount(feeLockAmount);
+
+  await unlockFee(testUser1);
+  await testUser1.refreshAmounts(AssetWallet.AFTER);
+  await checkAccountFeeLockAmount(0);
+});
+
+test("gassless- High-value swaps when successful are not charged txn fee or token timedout, but the percentage fee is charged", async () => {
+  const secondCurrency = await Assets.issueAssetToUser(
+    sudo,
+    defaultCurrencyValue,
+    sudo
+  );
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(secondCurrency, sudo, defaultCurrencyValue),
+    Assets.mintToken(secondCurrency, testUser1, defaultCurrencyValue),
+    Assets.mintNative(testUser1),
+    Sudo.sudoAs(
+      sudo,
+      Xyk.createPool(
+        firstCurrency,
+        defaultPoolVolumeValue,
+        secondCurrency,
+        defaultPoolVolumeValue
+      )
+    )
+  );
+
+  testUser1.addAsset(MGA_ASSET_ID);
+  testUser1.addAsset(firstCurrency);
+  testUser1.addAsset(secondCurrency);
+
+  const updateMgaTimeoutMetadata = await updateFeeLockMetadata(
+    sudo,
+    new BN(periodLength),
+    new BN(feeLockAmount),
+    thresholdValue,
+    [
+      [MGA_ASSET_ID, true],
+      [firstCurrency, true],
+    ]
+  );
+  await waitSudoOperataionSuccess(updateMgaTimeoutMetadata);
+
+  const sellAssetsValue = thresholdValue.add(new BN(5));
+
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+  await testUser1.sellAssets(firstCurrency, secondCurrency, sellAssetsValue);
+  await testUser1.refreshAmounts(AssetWallet.AFTER);
+
+  const userMgaFees = testUser1
+    .getAsset(MGA_ASSET_ID)
+    ?.amountAfter.free!.sub(
+      testUser1.getAsset(MGA_ASSET_ID)?.amountBefore.free!
+    );
+
+  expect(userMgaFees).bnEqual(new BN(0));
```

Comment:
plz check that reserved is also 0, ( cause is a high-value swap )

Reply:
I can not see where the reserved tokens are checked. @AlexChetverov  do you mind checking that the reserved tokens are 0 ?